src/axom/primal/geometry/Vector.hpp

```diff
@@ -540,8 +556,12 @@ template <typename T, int NDIMS>
 AXOM_HOST_DEVICE inline T Vector<T, NDIMS>::dot_product(const Vector<T, NDIMS>& u,
                                                         const Vector<T, NDIMS>& v)
 {
-  SLIC_ASSERT(u.dimension() == v.dimension());
-  return static_cast<T>(numerics::dot_product(u.data(), v.data(), NDIMS));
+  T res {};
+  for(int d = 0; d < NDIMS; ++d)
+  {
+    res += u[d] * v[d];
+  }
```

Comment:
Why did you make this change?

Reply:
It was an optimization. 

Vector knows the dimension (`NDIMS`) at compile time, but `numerics::dot_product` is an extra function call and uses a runtime variable for the loop bounds.

Vector::dot_product is called inside the barycenter and circumsphere computations which are in the inner loops of the Delaunay and ScatteredInterpolation computations.

Reply:
OK.  If the change is demanded by experience, great.